django/db/backends/oracle/schema.py

```diff
@@ -149,6 +149,9 @@ def _generate_temp_name(self, for_name):
         return self.normalize_name(for_name + "_" + suffix)
 
     def prepare_default(self, value):
+        if hasattr(value, 'as_sql'):
+            sql, params = self.prepare_param(value)
+            return sql % params
```

Comment:
Should some effort be made to `quote_value()` the params before construction?

Reply:
No, since this is a SQL expression like `NULL` or `CURRENT_TIMESTAMP` not a value.

Reply:
If a user did something unusual like `CharField(default=Value(";"))` would that not pass through this function and become a literal semicolon (which would probably not have been what the user expected)?

Reply:
@InvalidInterrupt is right, this should map `params` using `quote_value` just like it's done on other backends.

Reply:
After careful consideration, I must continue to object. See, this method is Oracle-specific, it isn't implemented nor called for other backends but Oracle. With the main reason that Oracle requires literal defaults. For that matter, quoting all values will simply not work, since it means quoting db expressions, which breaks the whole feature.

However, I do see the concern about `Values` expressions. I will add an exception for that. I hope that is satisfactory. If not I am open to solutions.